py/makeqstrdefs.py

```diff
@@ -22,6 +22,14 @@
 _MODE_COMPRESS = "compress"
 
 
+def is_c_source(fname):
+  return os.path.splitext(fname)[1] in [".c"]
+
+
+def is_cxx_source(fname):
+  return os.path.splitext(fname)[1] in [".cc", ".cp", ".cxx", ".cpp", ".CPP", ".c++", ".C"]
```

Comment:
Is `.C` for C++, or just C?

Would it be worth checking all case variants by doing `... .lower() in [".cc", ...]`?

Reply:
Yes, uppercase `.C` is for C++ files.
I'd keep it at that fixed list since that is also what compilers (gcc at least) do; and it would possibly get complicated (`.c` for C, `.C` for C++)

Example:

~~~c
#include <stdio.h>
int main(int argc, char ** argv) {
  printf(
  #ifdef __cplusplus
  "C++"
  #else
  "C"
  #endif
  );
}
~~~

~~~
$ ls -l
total 24
-rw-r--r-- 1 musteresel users 123 Mär 31 00:14 test.c
-rw-r--r-- 1 musteresel users 123 Mär 31 00:14 test.C
-rw-r--r-- 1 musteresel users 123 Mär 31 00:17 test.CC
-rw-r--r-- 1 musteresel users 123 Mär 31 00:14 test.cPp
-rw-r--r-- 1 musteresel users 123 Mär 31 00:15 test.CPP
-rw-r--r-- 1 musteresel users 123 Mär 31 00:17 test.CXX

$ for f in *; do echo "== ${f} =="; gcc $f -o the_test && ./the_test; echo; rm ./the_test &>/dev/null; done
== test.c ==
C
== test.C ==
C++
== test.CC ==
test.CC: file not recognized: file format not recognized
collect2: Fehler: ld gab 1 als Ende-Status zurück

== test.cPp ==
test.cPp: file not recognized: file format not recognized
collect2: Fehler: ld gab 1 als Ende-Status zurück

== test.CPP ==
C++
== test.CXX ==
test.CXX: file not recognized: file format not recognized
collect2: Fehler: ld gab 1 als Ende-Status zurück
~~~